### ISSUE TITLE
Inbox do not handle timeouts correctly

### ISSUE THREAD

**Issue by HCanber** [akkadotnet/akka.net#431]
For me Inbox_have_maximum_queue_size fails sometimes.

The test ends with this part that fails:
```c#
//The inbox should be empty now, so receiving should result in a timeout                
Assert.Throws<TimeoutException>(() =>
{
    var received=_inbox.Receive(TimeSpan.FromSeconds(1));
    Log.Error("Received "+received);
});
```
Inbox times out, but depending on where and how it does it different things happens.

`Receive` is implemented like this:
``` c#
var task=Receiver.Ask(new Get(DateTime.Now + timeout), Timeout.InfiniteTimeSpan);
if (task.Wait(timeout))
{
    return task.Result;
}
else
{
    var fmt = string.Format("Inbox {0} didn't received a response message in specified timeout {1}", 
        Receiver.Path, timeout);
    throw new TimeoutException(fmt);
}
```

If `task.Wait` times out the code returns `false`, and the expected exception is thrown and the test passes.
If the inbox times out internally (when handling the Kick message) the `Receiver` gets a `Status.Failure(new TimeoutException("Deadline passed"))` and the test failes.

Which is correct? To send a `Failure` message to receiver or raise a `TimeoutException`?

**Comment by annymsMthd** [comment]
Changes were merged in 6c7729907112785f63763198f818cbf6d5e7a0e5 to handle this. Please review and ensure this was the correct way to resolve this issue

**Comment by HCanber** [comment]
Looks ok to me.

**Comment by Horusiath** [comment]
Closed by #857
